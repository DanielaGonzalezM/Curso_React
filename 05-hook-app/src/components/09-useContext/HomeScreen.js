import React, { useContext } from "react";
import UserContext from "./UserContext";

const HomeScreen = () => {
  const {user} = useContext(UserContext);
  console.log(user)
  return (
    <div>
      <h3>HomeScreen</h3>
      <hr />
      <pre>
        {JSON.stringify(user,null,4)}
      </pre>
    </div>
  );
};

export default HomeScreen;
