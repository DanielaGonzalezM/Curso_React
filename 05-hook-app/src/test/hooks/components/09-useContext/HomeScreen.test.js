import{mount} from 'enzyme';
import HomeScreen from '../../../../components/09-useContext/HomeScreen';
import UserContext from '../../../../components/09-useContext/UserContext';



describe('Pruebas HomeScreen', () => { 
    const user={
        name: 'Daniela',
        email: 'daniela@gmail.com'
    }
    const wrapper = mount(
    <UserContext.Provider value={{user}}>
        <HomeScreen />
    </UserContext.Provider>
    
    );

    test('debe mostrarse correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot();

     })

 })