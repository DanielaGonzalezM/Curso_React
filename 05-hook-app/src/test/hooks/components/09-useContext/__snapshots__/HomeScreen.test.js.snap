// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas HomeScreen debe mostrarse correctamente 1`] = `
<div>
  <h3>
    HomeScreen
  </h3>
  <hr />
  <pre>
    {
    "name": "Daniela",
    "email": "daniela@gmail.com"
}
  </pre>
</div>
`;
